src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -36,6 +38,155 @@ Pane::Pane(const GUID& profile, const TermControl& control, const bool lastFocus
     }
 }
 
+// Method Description:
+// - Update the size of this pane. Resizes each of our columns so they have the
+//   same relative sizes, given the newSize.
+// - Because we're just manually setting the row/column sizes in pixels, we have
+//   to be told our new size, we can't just use our own OnSized event, because
+//   that _won't fire when we get smaller_.
+// Arguments:
+// - newSize: the amount of space that this pane has to fill now.
+// Return Value:
+// - <none>
+void Pane::ResizeContent(const Size& newSize)
+{
+    const auto width = newSize.Width;
+    const auto height = newSize.Height;
+
+    _CreateRowColDefinitions(newSize);
+
+    if (_splitState == SplitState::Vertical)
+    {
+        const auto paneSizes = _GetPaneSizes(width);
+
+        Size firstSize{ paneSizes.first, height };
+        Size secondSize{ paneSizes.second, height };
+        _firstChild->ResizeContent(firstSize);
+        _secondChild->ResizeContent(secondSize);
+    }
+    else if (_splitState == SplitState::Horizontal)
+    {
+        const auto paneSizes = _GetPaneSizes(height);
+
+        Size firstSize{ width, paneSizes.first };
+        Size secondSize{ width, paneSizes.second };
+        _firstChild->ResizeContent(firstSize);
+        _secondChild->ResizeContent(secondSize);
+    }
+}
+
+// Method Description:
+// - Adjust our child percentages to increase the size of one of our children
+//   and decrease the size of the other.
+// - Adjusts the separation amount by 5%
+// Arguments:
+// - direction: the direction to move our separator. If it's down or right,
+//   we'll be increasing the size of the first of our children. Else, we'll be
+//   decreasing the size of our first child.
+// Return Value:
+// - <none>
+void Pane::_DoResize(const Direction& direction)
+{
+    if (!DirectionMatchesSplit(direction, _splitState))
+    {
+        return;
+    }

```

Comment:
(This is gonna sound confusing. Bear with me.) So, as of now, we don't do anything if you try resizing not-the direction of the split? Example: if `SplitDirection::Horizontal` and `Direction::Left` --> do nothing?

Why not find the nearest valid separator and adjust the size of all its children? It's definitely more complex but doable and I feel like it makes sense.

Reply:
Or, what if it becomes more like "resize greater" and "resize lesser", and which one it impacts is based on how the current terminal's pane's split?

Reply:
Ignore my statement above... Started playing with it after hooking it up to some temp key-bindings and it does exactly what I'm thinking of haha.

Reply:
Lol yea np :P